src/main/java/com/extendedclip/deluxemenus/config/DeluxeMenusConfig.java

```diff
@@ -704,65 +705,42 @@ private Map<Integer, TreeMap<Integer, MenuItem>> loadMenuItems(FileConfiguration
-            DebugLevel.HIGHEST,
-            Level.WARNING,
-            "Material for item: " + key + " in menu: " + name + " is not valid!",
-            "Skipping item: " + key
+                DebugLevel.HIGHEST,
+                Level.WARNING,
+                "Material for item: " + key + " in menu: " + name + " is not valid!",
+                "Skipping item: " + key
         );
         continue;
       }
 
-      menuItem.setConfigMaterial(material);
-
-      // shield base color
-      menuItem.setConfigBaseColor(c.contains(currentPath + "base_color") && c.isString(currentPath + "base_color")
-          ? DyeColor.valueOf(c.getString(currentPath + "base_color", "WHITE").toUpperCase(Locale.getDefault()))
-          : null
-      );
-      // slot
-      menuItem.setSlot(c.getInt(currentPath + "slot", 0));
-      // amount
-      menuItem.setConfigAmount(c.getInt(currentPath + "amount", -1));
-      // dynamic amount
-      menuItem.setConfigDynamicAmount(c.getString(currentPath + "dynamic_amount", null));
-      // custom model data
-      menuItem.setCustomModelData(c.getString(currentPath + "model_data", null));
-      // displayname
-      menuItem.setConfigDisplayName(c.getString(currentPath + "display_name"));
-      // lore
-      if (c.isList(currentPath + "lore")) {
-        menuItem.setConfigLore(c.getStringList(currentPath + "lore"));
-      }
-      // rgb
-      menuItem.setRGB(c.getString(currentPath + "rgb", null));
-      // unbreakable
-      menuItem.setUnbreakable(c.getBoolean(currentPath + "unbreakable", false));
-      // update placeholders
-      menuItem.setUpdatePlaceholders(c.getBoolean(currentPath + "update", false));
-      // hide attributes
-      menuItem.setHideAttributes(c.getBoolean(currentPath + "hide_attributes", false));
-      // hide unbreakable
-      menuItem.setHideUnbreakable(c.getBoolean(currentPath + "hide_unbreakable", false));
-      // hide enchants
-      menuItem.setHideEnchants(c.getBoolean(currentPath + "hide_enchantments", false));
-      // hide potion effects
-      menuItem.setHidePotionEffects(c.getBoolean(currentPath + "hide_effects", false));
-      // nbt string
-      menuItem.setNbtString(c.getString(currentPath + "nbt_string", null));
-      // nbt int
-      menuItem.setNbtInt(c.getString(currentPath + "nbt_int", null));
-      // nbt strings
-      menuItem.setNbtStrings(c.getStringList(currentPath + "nbt_strings"));
-      // nbt ints
-      menuItem.setNbtInts(c.getStringList(currentPath + "nbt_ints"));
-      // priority
-      menuItem.setPriority(c.getInt(currentPath + "priority", 1));
+      MenuItemOptions.MenuItemOptionsBuilder builder = MenuItemOptions.builder()
+              .material(material)
+              .baseColor(c.contains(currentPath + "base_color") && c.isString(currentPath + "base_color")
+                      ? DyeColor.valueOf(c.getString(currentPath + "base_color", "WHITE").toUpperCase(Locale.getDefault()))
+                      : null)
```

Comment:
I think this can be simplified with an Optional
```java
Optional.ofNullable(c.getString(currentPath + "base_color"))
	.map(String::toUpperCase)
	.map(DyeColor::valueOf)
	.orElse(null)
```
I've just tested and a key like `base_color: ` has a null value. Unless they specify a color, e.g. `base_color: BLUE`, `MenuItemOptions#baseColor` will be null